### PR TITLE
Fix PDO::quote with string containing ASCII NUL character

### PR DESCRIPTION
**Pull request by v-kaywon**
Fix for https://github.com/Microsoft/msphpsql/issues/538

### CODE REVIEW

File: source/pdo_sqlsrv/pdo_dbh.cpp [hunk 1]
```diff
@@ -1463,7 +1463,7 @@ int pdo_sqlsrv_dbh_quote( _Inout_ pdo_dbh_t* dbh, _In_reads_(unquoted_len) const
         if ( encoding == SQLSRV_ENCODING_UTF8 ) {
             quotes_needed = 3;
         }
-        for ( size_t index = 0; index < unquoted_len && unquoted[ index ] != '\0'; ++index ) {
+        for ( size_t index = 0; index < unquoted_len; ++index ) {
```
Comment (yitam) [inline source/pdo_sqlsrv/pdo_dbh.cpp]: Is the argument `unquoted_len` always correct? What is `unquoted_len` if the string is empty? Or what if the input string is composed of only one null character, something like `'\0'`, or a series of null characters, something like `'\0\0\0\0'`?

File: test/functional/pdo_sqlsrv/pdo_538_quote_nul.phpt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+--FILE--
+<?php
+  
+require_once 'MsCommon.inc';
+
+try 
+{      
+    $connection = connect(); 
+    //$connection->setAttribute( PDO::SQLSRV_ATTR_DIRECT_QUERY, PDO::SQLSRV_ENCODING_SYSTEM );
+
```
Comment (yitam) [inline test/functional/pdo_sqlsrv/pdo_538_quote_nul.phpt]: Why did you comment this line? Have you tested this with PDO::SQLSRV_ENCODING_SYSTEM in Windows?

Comment (v-kaywon) [reply test/functional/pdo_sqlsrv/pdo_538_quote_nul.phpt]: I tested both ENCODING_UTF8 and ENCODING_SYSTEM and they are both OK no. I'll delete this line from the test.

File: source/pdo_sqlsrv/pdo_dbh.cpp (hunk 1 above)
Comment (v-kaywon) [reply source/pdo_sqlsrv/pdo_dbh.cpp]: From what I've tested, unquote_len seems to be always correct ( it was parsed out by the PDO extension using zend_parse_parameter. If the string is empty (""), unquoted_len is 0. If the input string is composed of only one null character, unquoted_len is 1, and for '\0\0\0\0' unquoted_len is 4.

File: test/functional/pdo_sqlsrv/pdo_538_quote_nul.phpt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+catch( PDOException $e ) {
+    die("Connection error: " . $e->getMessage());
+}
+?> 
+
+--EXPECT--
+Original: XX{NUL}XX
+Quoted:   'XX{NUL}XX'
```
Comment (lilgreenbird) [inline test/functional/pdo_sqlsrv/pdo_538_quote_nul.phpt]: what is this character at the EOF?

Comment (v-kaywon) [reply test/functional/pdo_sqlsrv/pdo_538_quote_nul.phpt]: no newline

**Pull request merged**